### DISCUSSION TITLE
Organisation and allocating of blocks

### DISCUSSION THREAD

**Discussion by mcctuxic** [tabemann/zeptoforth#165] [Q&A]
Hi,

by dividing blocks into ranges and assigning certain purposes to these ranges I want to try to organize blocks better.

Suppose I would use block 10 to 29 for general "configuration and setup of the system" and block 30 to 60 for general temporary experimention, and all blocks after that for really serious programming, which will bring peace and freedom to our 
world <cough> ;), would it put uneccessary load, waste of flash space, or other undesirable circumstances to my Pi Pico 2 W boards, if I first do
10 edit then 32 edit then 72 edit...?

Do the gaps in between really exist on  the flash physically - or are the number, nothing more than like names?
Does it matter, in which sequence blocks are edited?

Cheers!
mcc

**Comment by tabemann** [comment]
On Sun, Dec 8, 2024, 10:51 mcctuxic ***@***.***> wrote:

> Hi,
>
> by dividing blocks into ranges and assigning certain purposes to these
> ranges I want to try to organize blocks better.
>
> Suppose I would use block 10 to 29 for general "configuration and setup of
> the system" and block 30 to 60 for general temporary experimention, and all
> blocks after that for really serious programming, which will bring peace
> and freedom to our
> world ;), would it put uneccessary load, waste of flash space, or other
> undesirable circumstances to my Pi Pico 2 W boards, if I first do
> 10 edit then 32 edit then 72 edit...?
>
> Do the gaps in between really exist on the flash physically - or are the
> number, nothing more than like names?
> Does it matter, in which sequence blocks are edited?
>
> Cheers!
> mcc
>

In zeptoforth block IDs really are nothing but arbitrary ID codes - nothing
more. They can be created and edited in any order and there is no
disadvantage to having "holes" in their sequence (which do not waste any
space). Also, the only special value (that cannot be used) is $FFFFFFFF
(i.e. -1). The only other restriction is that blocks must not be used at
the same time as FAT32 filesystems in flash because the latter are
implemented on top of blocks.

Travis

>

**Comment by mcctuxic** [comment]
On 12/08 09:11, tabemann wrote:
> On Sun, Dec 8, 2024, 10:51 mcctuxic ***@***.***> wrote:
> 
> > Hi,
> >
> > by dividing blocks into ranges and assigning certain purposes to these
> > ranges I want to try to organize blocks better.
> >
> > Suppose I would use block 10 to 29 for general "configuration and setup of
> > the system" and block 30 to 60 for general temporary experimention, and all
> > blocks after that for really serious programming, which will bring peace
> > and freedom to our
> > world ;), would it put uneccessary load, waste of flash space, or other
> > undesirable circumstances to my Pi Pico 2 W boards, if I first do
> > 10 edit then 32 edit then 72 edit...?
> >
> > Do the gaps in between really exist on the flash physically - or are the
> > number, nothing more than like names?
> > Does it matter, in which sequence blocks are edited?
> >
> > Cheers!
> > mcc
> >
> 
> In zeptoforth block IDs really are nothing but arbitrary ID codes - nothing
> more. They can be created and edited in any order and there is no
> disadvantage to having "holes" in their sequence (which do not waste any
> space). Also, the only special value (that cannot be used) is $FFFFFFFF
> (i.e. -1). The only other restriction is that blocks must not be used at
> the same time as FAT32 filesystems in flash because the latter are
> implemented on top of blocks.
> 
> Travis
> 
> >
> 

Hi Travis,

thanks for the info! That makes things a lot easier...

Cheers!
mcc
